src/ai/backend/storage/weka/weka_client.py

```diff
@@ -0,0 +1,388 @@
+        if len(data["data"]) == 0:
+            raise WekaNotFoundError
+        return [WekaFs.from_json(fs_json) for fs_json in data["data"]]
+
+    @error_handler
+    async def get_fs(self, fs_uid: str) -> WekaFs:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(sess, "GET", f"/fileSystems/{fs_uid}")
+            data = await response.json()
+        if data.get("data") is None:
+            raise WekaNotFoundError
+        return WekaFs.from_json(data["data"])
+
+    @error_handler
+    async def list_quotas(self, fs_uid: str) -> Iterable[WekaQuota]:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota",
+            )
+            data = await response.json()
+        if isinstance(data['data'], list):
+            return [
+                WekaQuota.from_json(quota_info["quota_id"], quota_info)
+                for quota_info in data["data"]
+            ]
+        else:
+            return [
+                WekaQuota.from_json(quota_id, data["data"][quota_id])
+                for quota_id in data["data"].keys()
+            ]
+
+    @error_handler
+    async def get_quota(self, fs_uid: str, inode_id: int) -> WekaQuota:
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "GET",
+                f"/fileSystems/{fs_uid}/quota/{inode_id}",
+            )
+            data = await response.json()
+        if len(data["data"].keys()) == 0:
+            raise WekaNotFoundError
+        quota_id = data["data"].keys()[0]
+        return WekaQuota.from_json(quota_id, data["data"][quota_id])
+
+    @error_handler
+    async def set_quota(
+        self,
+        fs_uid: str,
+        inode_id: int,
+        soft_limit: BinarySize,
+        hard_limit: BinarySize,
+    ) -> WekaQuota:
+        def _format_size(s: BinarySize) -> str:
+            ss = str(s)
+            if ss.endswith("bytes"):
+                return ss.replace("bytes", "B").replace(" ", "")
+            else:
+                return ss.replace(" ", "")
+
+        body = {
+            "grace_seconds": None,
+            "soft_limit_bytes": _format_size(soft_limit),
+            "hard_limit_bytes": _format_size(hard_limit),
+        }
+
+        async with aiohttp.ClientSession(
+            base_url=self.api_endpoint,
+        ) as sess:
+            response = await self._build_request(
+                sess,
+                "PUT",
+                f"/fileSystems/{fs_uid}/quota/{inode_id}",
+                body,
+            )
+            data = await response.json()
+
+        quota_id = data["data"].keys()[0]
+        return WekaQuota(
```

Comment:
Since we adopted "directory-based" quota strategy from weka integration, quota_id received from weka API server would be only one.
There's no need to parsing "quota_id" from response. Instead, Let's just leave quota_id as None.(or It may be better to get removed from member variable in [WekaQuota](https://github.com/lablup/backend.ai/pull/443/files#diff-36cfe4b50d0a1c978df8cb77d6e45162c6df438ed66997fc94e8d231218b00cbR21-R26) class).

Reply:
In that case, `quota_id` key in [L294-L297 ](https://github.com/lablup/backend.ai/pull/443/files#diff-36cfe4b50d0a1c978df8cb77d6e45162c6df438ed66997fc94e8d231218b00cbR294-R297)should be removed likewise.

Reply:
Writing `weka_client.py` like is to easily adapt and expand its code to other projects. Of course, ignoring `quota_id`  also will be a good if we're going to use `WekaAPIClient` class only in storage-proxy, but I don't think that approach won't be a extensible one. Please add your opinion about this.

Reply:
:( my bad. There was a typo on the code. Fixed, so it should work now.